src/ImageSharp/Formats/Bmp/BmpDecoderCore.cs

```diff
@@ -1422,6 +1422,16 @@ private int ReadImageHeaders(BufferedReadStream stream, out bool inverted, out b
         this.ReadFileHeader(stream);
         this.ReadInfoHeader(stream);
 
+        // BMPs with negative-or-zero width are invalid. Also, reject extremely wide images
+        // to keep the math sane. And reject int.MinValue as a height because you can't
+        // get its absolute value (because -int.MinValue is one more than int.MaxValue).
+        const int k64KWidth = 65535;
```

Comment:
should this not be a configurable value? I can see use cases where someone might want to load very large images... they would be responsible to ensure they have enough memory. This is a sensible default looks fine but allowing a user to bypass where they understand the danger etc.... I can also see use cases in the imagesharp.web world where we might even want even lower limits to prevent bad actors blowing out memory etc.

We might even want to consider a more generic max width/max height set of rules that can be applied to all image formats uniformly.

Reply:
True, updated.